### PR TITLE
[nethost] Bump to 8.0.0

### PR DESCRIPTION
**Pull request by julianxhokaxhiu**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

---

Bump the following ports:
- nethost: 7.0.0 -> 8.0.0

See also: https://devblogs.microsoft.com/dotnet/announcing-dotnet-8/

### CODE REVIEW

File: ports/nethost/portfile.cmake [hunk 1]
```diff
@@ -1,19 +1,20 @@
-set(COMMIT_HASH v7.0.0)
+set(PRODUCT_VERSION "8.0.0")
+set(COMMIT_HASH "v${PRODUCT_VERSION}")
```
Comment (MonicaLiu0311) [inline ports/nethost/portfile.cmake:2]:
```suggestion
```

File: ports/nethost/portfile.cmake [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO dotnet/runtime
     REF ${COMMIT_HASH}
```
Comment (MonicaLiu0311) [inline ports/nethost/portfile.cmake:7]:
```suggestion
    REF "v${VERSION}"
```
${VERSION} is obtained from vcpkg.json.

**Comment by MonicaLiu0311** [comment]
-`file(INSTALL "${SOURCE_PATH}/LICENSE.TXT" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)`
+`vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE.TXT")`

**Comment by julianxhokaxhiu** [comment]
Thanks @MonicaLiu0311 I applied all the suggestions. Feel free to review :)

**Review by MonicaLiu0311 — DISMISSED** [review]

File: ports/nethost/portfile.cmake [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
 )
 
-set(PRODUCT_VERSION "7.0.0")
-
 if(NOT VCPKG_TARGET_IS_WINDOWS)
-  execute_process(COMMAND sh -c "mkdir -p ${SOURCE_PATH}/artifacts/obj && ${SOURCE_PATH}/eng/native/version/copy_version_files.sh")
+  execute_process(COMMAND "sh" "-c" "mkdir -p ${SOURCE_PATH}/artifacts/obj && ${SOURCE_PATH}/eng/native/version/copy_version_files.sh")
+else()
+  execute_process(COMMAND "cmd.exe" "/c" "mkdir artifacts\\obj && eng\\native\\version\\copy_version_files.cmd" WORKING_DIRECTORY "${SOURCE_PATH}")
 endif()
```
Comment (dg0yt) [inline ports/nethost/portfile.cmake:15]:
~~~suggestion
file(MAKE_DIRECTORY "${SOURCE_PATH}/artifacts/obj")
set(copy_version_files  sh -c "${SOURCE_PATH}/eng/native/version/copy_version_files.sh")
if(VCPKG_HOST_IS_WINDOWS)
  set(copy_version_files  cmd /C "eng\\native\\version\\copy_version_files.cmd")
endif()
vcpkg_execute_required_process(
  COMMAND ${copy_version_files}
  WORKING_DIRECTORY "${SOURCE_PATH}"
  LOGNAME "copy_version_files-${TARGET_TRIPLET}"
)
~~~
Key change is `VCPKG_HOST_IS_WINDOWS`.
But everything else is nice to have, too.

**Review by dg0yt — CHANGES_REQUESTED** [review]

**Comment by julianxhokaxhiu** [comment]
Thanks @dg0yt applied your suggestions as well. Feel free to review :)

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-windows` (header files found):
```
The package nethost can be used via CMake:

    find_package(unofficial-nethost CONFIG REQUIRED)
    target_link_libraries(main PRIVATE unofficial::nethost::nethost)
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Pull request merged**